converters/toZigbee.js

```diff
@@ -147,6 +147,48 @@ const converters = {
             await entity.read('genOnOff', ['onOff']);
         },
     },
+    TYZB01_on_off: {
+        key: ['state', 'transition'],
+        convertSet: async (entity, key, value, meta) => {
+            const result = await converters.on_off.convertSet(entity, key, value, meta);
```

Comment:
I would expect the `converters.on_off.convertSet` only be called on a toggle. Now always first an on/off command is send followed by the onWithTimedOff of this converter. Is this correct?

Reply:
Yes. This is intended. And just to clarify because I might have not made it clear in the PR's description.  The command `set: { state: "ON", transition: 10 }` means - turn on and then turn off after 10 seconds. The modules misinterpret the Zigbee **onWithTimedOff** command in **onOff** cluster. They understand the command as follows:
* if the switch is on (onOff == 1) and received `onWithTimedOff: { ctrlbits: (no-matter), ontime: (no-matter), offwaittime: (x) }` command, turn the switch off after x seconds
* if the switch is off (onOff == 0) and received `onWithTimedOff: { ctrlbits: (no-matter), ontime: (x), offwaittime: (no-matter) }`, turn the switch on after x seconds

Reply:
Ah sorry, now I understand this MR. The `transition` doesn't fit in this case, `time_in_seconds` like you initially did suits better.